kindedj/src/main/java/io/kindedj/Hk.java

```diff
@@ -16,5 +16,14 @@
 
 package io.kindedj;
 
-public interface HK<F, A> {
+/**
+ * Hk, or higher kind, represents a value that wraps another value. It would be equivalent to F&lt;A&gt; if Java supported higher kinds.
+ * <br/><br/>
+ * To construct higher kinds of arity 2+ you have to nest Hk into its witness type parameter.
+ * For example, to represent F&lt;G&lt;H&lt;A&gt;&gt;&gt; you have to create an interface or type alias Hk&lt;Hk&lt;Hk&lt;F, G&gt;, H&gt;, A&gt;
```

Comment:
s/`F<G<H<A>>>`/`F<A, B, C>`
(`F<G<H<A>>>`actually means that F is only of arity 1)